docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,88 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies changes to the auth and bank modules to implement vesting accounts for the Cosmos Hub. 
+The requirements for this vesting account is that it should be capable of being initialized during genesis with
+a starting balance X coins and a vesting blocktime T. The owner of this account should be able to delegate to validators 
+and vote with locked coins, however they cannot send locked coins to other accounts until those coins have been unlocked. 
+The vesting account should also be able to spend any coins it receives from other users or from fees/inflation rewards. 
```

Comment:
Old spec was not able to spend coins gained from fees/inflation. New spec will allow vesting account to do so.

Reply:
wait this is kind of seperate, the ability to spend coins received through the fee-distribution mechanism is independent of the vesting account because the fee-distribution mechanism allows you with withdraw fees to a seperate account - we should keep these distinct and not make the vesting account have special logic for this

Reply:
Right, the purpose of `ReceivedCoins` is to makes sure that funds sent to the VestingAccount after initialization are spendable immediately.